### PR TITLE
MINOR: Add verification step for Streams archetype to Jenkins build

### PR DESCRIPTION
**Pull request by mjsax**
Updates `./jenkins.sh` to build stream archetype and install it in local maven cache. Afterward, archetype is used to create a new maven project and maven project is compiled for verification.

### CODE REVIEW

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @vvcephei @ableegoldman (\cc @ijuma)

File: streams/quickstart/java/src/main/resources/archetype-resources/src/main/java/LineSplit.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public static void main(String[] args) throws Exception {
 
         final StreamsBuilder builder = new StreamsBuilder();
 
-        builder.stream("streams-plaintext-input")
+        builder.<String, String>stream("streams-plaintext-input")
```
Comment (mjsax) [inline streams/quickstart/java/src/main/resources/archetype-resources/src/main/java/LineSplit.java]: This fix should actually be provided by https://github.com/apache/kafka/pull/6194 -- I included it here to let the build pass.

File: jenkins.sh [hunk 2]
```diff
@@ -28,3 +28,45 @@
 ./gradlew unitTest integrationTest \
     --profile --no-daemon --continue -PtestLoggingEvents=started,passed,skipped,failed "$@" \
     || { echo 'Test steps failed'; exit 1; }
+
+# Verify that Kafka Streams archetype compiles
+if [ $JAVA_HOME = "/home/jenkins/tools/java/latest11" ] ; then
```
Comment (mjsax) [inline jenkins.sh:33]: The created archetype, only works for Java8 -- we would need to create a different archetype for Java11.

What we could do, is to include the Java11 setting as comments in the archetype (thoughts)?

**Comment by mjsax** [comment]
Java8 failed as expected:
```
21:35:30 [ERROR] COMPILATION ERROR : 
21:35:30 [INFO] -------------------------------------------------------------
21:35:30 [ERROR] /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk8-scala2.11@2/streams/quickstart/test-streams-archetype/streams.examples/src/main/java/myapps/LineSplit.java:[48] 
21:35:30 	.flatMapValues(value -> Arrays.asList(value.split("\\W+")))
21:35:30 	                                            ^^^^^
21:35:30 The method split(String) is undefined for the type Object
21:35:30 1 problem (1 error)
```

File: jenkins.sh (hunk 2 above)
Comment (vvcephei) [reply jenkins.sh:33]: We could do that, or we can just publish another archetype for use with Java 11+. Then, when we drop support for Java 8, it's as simple as just deleting the Java 8 archetype.

**Review by vvcephei — COMMENTED** [review]
Hey @mjsax , thanks for taking care of this!

I've responded to your question with my $0.02.

File: jenkins.sh (hunk 2 above)
Comment (mjsax) [reply jenkins.sh:33]: Not sure how we could have two archetypes at the same time? I guess we would need to encode the Java version in the archetype name?
```
-DarchetypeGroupId=org.apache.kafka
-DarchetypeArtifactId=streams-quickstart-java   <--- change this to `streams-quickstart-java-8` and `streams-quickstart-java-11` ?
-DarchetypeVersion=$version 
```

Comment (vvcephei) [reply jenkins.sh:33]: Yep, that would be the way to do it.

Comment (mjsax) [reply jenkins.sh:33]: @guozhangwang Can/should we do this in this PR? Or would this need a KIP? Or just create a JIRA for a follow up PR?

File: docs/streams/tutorial.html [hunk 3]
```diff
@@ -74,8 +74,12 @@ <h4><a id="tutorial_maven_setup" href="#tutorial_maven_setup">Setting up a Maven
     </pre>
 
     <p>
-        The <code>pom.xml</code> file included in the project already has the Streams dependency defined,
-        and there are already several example programs written with Streams library under <code>src/main/java</code>.
+        The <code>pom.xml</code> file included in the project already has the Streams dependency defined.
+	Note, that the generated <code>pom.xml</code> targets Java 8, and does not work with higher Java versions.
```
Comment (guozhangwang) [inline docs/streams/tutorial.html]: nit: one more indent.

File: jenkins.sh (hunk 2 above)
Comment (guozhangwang) [reply jenkins.sh:33]: We can create a JIRA ticket for a follow-up PR (this would need a KIP, yes).

**Review by guozhangwang — APPROVED** [review]
LGTM. Please feel free to merge after jenkins.

File: jenkins.sh (hunk 2 above)
Comment (mjsax) [reply jenkins.sh:33]: Created https://issues.apache.org/jira/browse/KAFKA-8116

**Comment by mjsax** [comment]
Rebased to get the fix from #6194.

**Comment by mjsax** [comment]
Java11 failed with known flaky tests.

Java8 failed with (https://builds.apache.org/job/kafka-pr-jdk8-scala2.11/20398/console):
```
22:29:21 org.apache.kafka.streams.KafkaStreamsTest.statefulTopologyShouldCreateStateDirectory failed, log available in /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk8-scala2.11/streams/build/reports/testOutput/org.apache.kafka.streams.KafkaStreamsTest.statefulTopologyShouldCreateStateDirectory.test.stdout
22:29:21 
22:29:21 org.apache.kafka.streams.KafkaStreamsTest > statefulTopologyShouldCreateStateDirectory FAILED
22:29:21     java.io.UncheckedIOException: java.nio.file.NoSuchFileException: /tmp/kafka-9Bd3Q/appId/1_0/rocksdb/statefulTopologyShouldCreateStateDirectory-counts/000007.dbtmp
22:29:21         at java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:88)
22:29:21         at java.nio.file.FileTreeIterator.hasNext(FileTreeIterator.java:104)
22:29:21         at java.util.Iterator.forEachRemaining(Iterator.java:115)
22:29:21         at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
22:29:21         at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
22:29:21         at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
22:29:21         at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
22:29:21         at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
22:29:21         at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
22:29:21         at org.apache.kafka.streams.KafkaStreamsTest.startStreamsAndCheckDirExists(KafkaStreamsTest.java:744)
22:29:21         at org.apache.kafka.streams.KafkaStreamsTest.statefulTopologyShouldCreateStateDirectory(KafkaStreamsTest.java:661)
22:29:21 
22:29:21         Caused by:
22:29:21         java.nio.file.NoSuchFileException: /tmp/kafka-9Bd3Q/appId/1_0/rocksdb/statefulTopologyShouldCreateStateDirectory-counts/000007.dbtmp
22:29:21             at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
22:29:21             at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
22:29:21             at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
22:29:21             at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
22:29:21             at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
22:29:21             at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
22:29:21             at java.nio.file.Files.readAttributes(Files.java:1737)
22:29:21             at java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:219)
22:29:21             at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
22:29:21             at java.nio.file.FileTreeWalker.next(FileTreeWalker.java:372)
22:29:21             at java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:84)
22:29:21             ... 10 more
```

Is this an environmental issue or indicates a flaky test?

Retest this please.

**Comment by mjsax** [comment]
Java 11 failed. Created ticket.
Java 8 passed.

Retest this please.

**Review by bbejeck — APPROVED** [review]
LGTM

**Comment by bbejeck** [comment]
Merged #6431 into trunk.

**Pull request merged**